File: frontend/src/api/generateSchema.ts
import { apiFetch } from './client';

interface GenerateSchemaRequest {
  description: string;
}

interface GenerateSchemaResponse {
  schema_name: string;
  schema_body: Record<string, unknown>;
}

export async function generateSchema(description: string): Promise<GenerateSchemaResponse> {
  return apiFetch<GenerateSchemaResponse>('/generate-schema', {
    method: 'POST',
    body: JSON.stringify({ description } satisfies GenerateSchemaRequest),
  });
}

File: src/open_eval/routers/schemas/generate_schema.py
"""Pydantic schemas for the AI schema generation endpoint."""

from pydantic import BaseModel


class GenerateSchemaRequest(BaseModel):
    """Request body for AI-powered JSON schema generation."""

    description: str


class GenerateSchemaResponse(BaseModel):
    """Response body with generated JSON schema."""

    schema_name: str
    schema_body: dict
